File: python/nt_solutions/segmentation_3_snakes_levelset/exo7.py
plt.figure(figsize=(10,5))
Y,X = np.meshgrid(np.arange(1,n+1), np.arange(1,n+1))
k = 4 #number of circles
r = .3*n/k
phi0 = np.zeros([n,n])+np.float("inf")

for i in range(1,k+1):
    for j in range(1,k+1):
        c = (np.asarray([i,j]) - 1)*(n/k) + (n/k)*.5
        phi0 = np.minimum(phi0,np.sqrt(abs(X-c[0])**2 + abs(Y-c[1])**2) - r)
        
plt.subplot(1,2,1)
plot_levelset(phi0,0)
plt.subplot(1,2,2)
plot_levelset(phi0, 0, f0)

File: r/nt_solutions/segmentation_3_snakes_levelset/exo7.R





grid <- meshgrid_2d(1:n, 1:n)
Y <- grid$X ; X <- grid$Y
k <- 4 #number of circles
r <- .3*n/k
phi0 <- array(Inf, c(n,n))

for (i in 1:k){
  for (j in 1:k){
    c <- (c(i,j) - 1)*(n/k) + (n/k)*.5
    phi0 <- pmin(phi0,
                 sqrt(abs(X-c[1])**2 + abs(Y-c[2])**2) - r)
  }
}

par(mfrow=c(1,2))

plot_levelset(phi0,lw=2)

plot_levelset(phi0, f0, lw=2)